### DISCUSSION TITLE
Image disk/memory cache

### DISCUSSION THREAD

**Discussion by ippocratis** [savvasdalkitsis/uhuruphotos-android#121] [Ideas]
Again , idk if the api even provide a way of doting this :
Add an option to download and cache all assert thumbnails at once .
The app is already fast at loading uncached images but for devices with no storage issues it would be great to have all thumbnails predownloaded. 
This will make the scrolling and the ui feeling more fluid.

**Comment by savvasdalkitsis** [comment]
This sounds doable and a good idea. i will look at implementing it soon

**Comment by savvasdalkitsis** [comment]
This is now implemented by #123

**Comment by ippocratis** [comment]
I've run a pre-cahe
It reached 100%
Disabled internet so I could check they are not loading on-the-fly
but no thumbnails are shown if I scroll on the time feed
It supposed to pre-load all thumbnails in my entire feed right?
Do I miss something?

**Reply by savvasdalkitsis** [reply]
It's attempting to fetch all images so they are cached in memory and disk. Of course if you have more photos that can fit in the disk cache, then the first ones to be preached will be evicted.

Try increasing your disk cache size perhaps?

I also know that I'm loading them from oldest to newest which kinda defeats the purpose. I should probably reverse the order so that the oldest photos are guaranteed to be cached? Not sure what the right answer here is tbh

**Reply by ippocratis** [reply]
My other related settings are:

Image disk cache:2000mb
Curently used :40mb

Image memory cache:2000mb
Currently used:104mb

Video disk cacje:2000mb
Currently used:15mb

Also from the andeoid app info page,  under storage&cache it states that the app size is 108mb with 24mb user data and 112mb cache

So hitting the limits is not the case

Maybe I could run another precache with logs enabled? 

Might those be of any help?

Anyway I really don't want t to interfere with feature that are maybe lower in the priority list

**Reply by savvasdalkitsis** [reply]
Hm that's weird indeed. I'll try to replicate and fix it

**Reply by savvasdalkitsis** [reply]
found the bug. will be fixed in the next release

**Reply by ippocratis** [reply]
OK we're almost there :-)

Preaching indeed reached 100% and started to cache images/videos from older to newer photos 
*But even tho the image/video disk/memory cache did not hit the limits (reached about 500mb on a 2000mb limit) , not all asserts where cached*
I'll have to perform another precaching though coz It was a fresh install and I forgot to make a full sync first
not sure if its related but I want to make it right so I can provide legit feedback
Again if you need any logs please ask


Some side notes if you may:
If there wasn't a storage limit the "direction" of caching (past to present  or pressent to past) wouldn't matter but since there is a Max 2000mb limit  I think I would prefer the most recent asserts cached first
Do you really need the storage limitation?
I remember when logging went huge in size there where a performance drop in the app 
So maybe there is a real reason for that


Again
Thanks for maintaining this amazing app
Sorry if I ask too much :-)
And keep up

**Reply by savvasdalkitsis** [reply]
Next version will allow you to set the mem and disk caches up to your device max so at least that will not be a problem anymore
